File: test-suite/java/security/RandomSecurityBuggy.java
import java.util.Random;
import java.util.SplittableRandom;
import java.util.concurrent.ThreadLocalRandom;

class RandomSecurityBuggy {
    private final Random rng = new Random();

    String resetToken() {
        long token = rng.nextLong();
        return Long.toString(token, 36);
    }

    String sessionSecret() {
        Random local = new Random(System.currentTimeMillis());
        return "sess_" + Long.toString(local.nextLong(), 36);
    }

    String csrfNonce() {
        String nonce = Long.toString(System.currentTimeMillis(), 36);
        return "csrf_" + nonce;
    }

    String apiKey() {
        return "ak_" + ThreadLocalRandom.current().nextInt(1_000_000);
    }

    String oneTimePassword() {
        int otp = new SplittableRandom().nextInt(100000, 1000000);
        return String.format("%06d", otp);
    }

    String inviteToken() {
        return Long.toString((long) (Math.random() * Long.MAX_VALUE), 36);
    }
}

File: test-suite/csharp/security/RandomSecurityBuggy.cs
using System;
using System.Diagnostics;

public sealed class RandomSecurityBuggy
{
    private readonly Random rng = new Random();

    public string ResetToken()
    {
        long token = rng.NextInt64();
        return token.ToString("x");
    }

    public string SessionSecret()
    {
        var local = new Random(Environment.TickCount);
        return "sess_" + local.NextInt64().ToString("x");
    }

    public string CsrfNonce()
    {
        string nonce = DateTime.UtcNow.Ticks.ToString("x");
        return "csrf_" + nonce;
    }

    public string ApiKey()
    {
        return "ak_" + Random.Shared.Next(1_000_000);
    }

    public string OneTimePassword()
    {
        return new Random().Next(100000, 999999).ToString();
    }

    public string InviteToken()
    {
        return Guid.NewGuid().ToString("N");
    }

    public string RecoverySecret()
    {
        return "r_" + Process.GetCurrentProcess().Id.ToString("x");
    }
}
